src/mito2/src/manifest/tests/checkpoint.rs

```diff
@@ -159,6 +160,53 @@ async fn manager_with_checkpoint_distance_1() {
     assert_eq!(10, manager.manifest().await.manifest_version);
 }
 
+#[tokio::test]
+async fn test_corrupted_data_causing_checksum_error() {
+    // Initialize manager
+    common_telemetry::init_default_ut_logging();
+    let (_env, manager) = build_manager(1, CompressionType::Uncompressed).await;
+
+    // Apply actions
+    for _ in 0..10 {
+        manager.update(nop_action()).await.unwrap();
+    }
+
+    // Check if there is a checkpoint
+    assert!(manager
+        .store()
+        .await
+        .load_last_checkpoint()
+        .await
+        .unwrap()
+        .is_some());
+
+    // Corrupt the last checkpoint data
+    let mut corrupted_bytes = manager
+        .store()
+        .await
+        .read_file(&manager.store().await.last_checkpoint_path())
+        .await
+        .unwrap();
+    corrupted_bytes[0] ^= 1;
+
+    // Overwrite the latest checkpoint data
+    let _ = manager
+        .store()
+        .await
+        .write_last_checkpoint(9, &corrupted_bytes)
+        .await
+        .unwrap();
```

Comment:
Do we need this method? We could write to the object store using the last_checkpoint_path directly.